src/aead/gcm.rs

```diff
@@ -84,16 +89,33 @@ impl Key {
 
 pub struct Context {
     inner: ContextInner,
+    aad_len: BitLength<u64>,
+    in_out_len: BitLength<u64>,
     cpu_features: cpu::Features,
 }
 
 impl Context {
-    pub(crate) fn new(key: &Key, aad: Aad<&[u8]>, cpu_features: cpu::Features) -> Self {
+    pub(crate) fn new(
+        key: &Key,
+        aad: Aad<&[u8]>,
+        in_out_len: usize,
+        cpu_features: cpu::Features,
+    ) -> Result<Self, error::Unspecified> {
+        if in_out_len > aes_gcm::MAX_IN_OUT_LEN {
+            return Err(error::Unspecified);
+        }
```

Comment:
This code path only occurs with a (practically) impossibly-long input, so we can't test it.